integration/v3_grpc_test.go

```diff
@@ -193,3 +195,135 @@ func TestV3DeleteRange(t *testing.T) {
 		clus.Terminate(t)
 	}
 }
+
+// TestV3WatchFromCurrentRevision tests Watch APIs from current revision.
+func TestV3WatchFromCurrentRevision(t *testing.T) {
+	testKey, testKeyL, testKey2 := "foo", "fooLong", "helloworld"
+	testVal := "bar"
+
+	tests := []struct {
+		putKeys        []string
+		watchRequest   *pb.WatchRequest
+		headerRevision int64
+		createdN       int
+		wevents        []*storagepb.Event
+	}{
+		// watch the key, matching
+		{
+			[]string{testKey},
+			&pb.WatchRequest{CreateRequest: &pb.WatchCreateRequest{Key: []byte(testKey)}},
+			1,
+			1,
+			[]*storagepb.Event{
+				&storagepb.Event{
+					Kv: &storagepb.KeyValue{Key: []byte(testKey), Value: []byte(testVal), CreateRevision: 1, ModRevision: 1, Version: 1},
+				},
+			},
+		},
+		// watch the key, non-matching
+		{
+			[]string{testKey},
+			&pb.WatchRequest{CreateRequest: &pb.WatchCreateRequest{Key: []byte(testKey2)}},
+			1,
+			0,
```

Comment:
Yes that's correct. Will fix. somehow it wasn't returning CreatedRequest. Need to investigate what I am doing wrong.

Thanks,